libipl/p10/ipl0.C

```diff
@@ -292,6 +296,37 @@ static bool update_genesis_hwas_state(void)
 		}
 	}
 
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+		if(!pdbg_target_get_attribute(
+			clock_target, "ATTR_POSITION", 2, 1, &clk_pos)) {
+
+			ipl_log(IPL_ERROR, "Attribute ATTR_POSITION read failed"
+				" for clock '%s' \n", pdbg_target_path(clock_target));
+			ipl_error_callback(IPL_ERR_ATTR_ACCESS);
+			// IPL need to be failed if this step is failed for any clock,
+			// since this clock cannot be marked as functional
+			return false;
+		}
+
+		status = pdbg_target_probe(clock_target);
+		if(status != PDBG_TARGET_ENABLED) {
+			ipl_log(IPL_ERROR, "clock '%s' is not operational, pdbg status = %d\n",
+				pdbg_target_path(clock_target), status);
+
+			ffdcs.push_back(std::make_pair("PDBG_STATUS", std::to_string(status)));
+			ffdcs.push_back(std::make_pair("FAIL_TYPE", "CHIP_NOT_OPERATIONAL"));
+			ipl_plat_clock_error_handler(ffdcs, clk_pos);
+			// IPL need to be failed if any clock is not operational
```

Comment:
should we deconfigure the clock continue re-ipl incase redudnavcy enabled systems?

Reply:
No. In openbmc system there is no true clock redundancy. As per current desing, we will fail if any clock failed. Later we will enable integrated sparing. But for this we need more infrastructure changes

Reply:
ok can you add some additional comment in the code for future refernce.,